{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3a905086-232b-42ef-9b68-89fb5ea5b3ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "import synapseclient\n",
    "from synapseclient import Project, File, Folder\n",
    "from synapseclient import Schema, Column, Table, Row, RowSet, as_table_columns\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4225e1b5-5114-4731-86b8-4b4962937998",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome, Victor Baham!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "syn = synapseclient.Synapse()\n",
    "syn.login(authToken='eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIiwibW9kaWZ5Il0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTcyOTI3MjUwMiwiaWF0IjoxNzI5MjcyNTAyLCJqdGkiOiIxMjkyMyIsInN1YiI6IjM0NDk0MDQifQ.cCgWsjIVBw5wwVxhKqTveKl_dhgswQQ-1fn_LjCDlUCgG38lNJuHac-HE29tgFHIP1k2-2uN-LTzpSd5vm8XH9A5P8Girmdz1XmDCVRcJmS6UK8j5owyo6mTC9Ca6c93fPUnmgJPCUzhE4yiLT2RQBmlqFLRm20EvpT2QqoZLCGxNlOMSKmyuUAHqV0D2imFc32vv4OTVRPTB5_0WxZoL30vAHfPWspcxfBgmMkAIlbLhUzoKrXwXrT8etuEBfLj8P0FSnz1tBO5H1BjVADAYlN6J_sOwI0cT56Q8X88hmmNlcAKN5lI6Zfe8PwHS7JNcF3yi7s4lKm7e_tI9ga3oQ')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2271fe2d-787a-4155-bc4e-8d50d770f537",
   "metadata": {},
   "source": [
    "#### Get individual and biospecimen metadata for the MSBB and ROSMAP cohorts from the NPS-AD study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0ec902a3-1bd6-4879-a5b9-5307cb9189b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Component_x</th>\n",
       "      <th>individualID</th>\n",
       "      <th>cohort</th>\n",
       "      <th>species</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>yearsEducation</th>\n",
       "      <th>apoeGenotype</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>nucleicAcidSource</th>\n",
       "      <th>cellType</th>\n",
       "      <th>samplingAge</th>\n",
       "      <th>samplingAgeUnits</th>\n",
       "      <th>sampleStatus</th>\n",
       "      <th>fastingState</th>\n",
       "      <th>visitNumber</th>\n",
       "      <th>assay</th>\n",
       "      <th>Id</th>\n",
       "      <th>entityId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>AMPAD_MSSM_0000029668</td>\n",
       "      <td>MSBB</td>\n",
       "      <td>Human</td>\n",
       "      <td>male</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>337399ff-c8cd-4ee2-a87c-11488db02558</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>AMPAD_MSSM_00000M1061</td>\n",
       "      <td>MSBB</td>\n",
       "      <td>Human</td>\n",
       "      <td>male</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>bf1ad418-d4bc-492a-8426-b66c8fe7e9a2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>AMPAD_MSSM_00000M1062</td>\n",
       "      <td>MSBB</td>\n",
       "      <td>Human</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>50757002-0330-441c-9e61-2b481494bb09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>AMPAD_MSSM_0000010001</td>\n",
       "      <td>MSBB</td>\n",
       "      <td>Human</td>\n",
       "      <td>female</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>1593a784-36bf-449d-a476-2c6d29f11858</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>AMPAD_MSSM_0000010031</td>\n",
       "      <td>MSBB</td>\n",
       "      <td>Human</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>c2514bf6-2229-4a55-a3fe-65f7a673ddd1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1489</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>R5508487</td>\n",
       "      <td>ROSMAP</td>\n",
       "      <td>Human</td>\n",
       "      <td>male</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>2b02ebb9-a0e9-4883-9e2e-7ba7350499f7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1490</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>R9380629</td>\n",
       "      <td>ROSMAP</td>\n",
       "      <td>Human</td>\n",
       "      <td>female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>b9ad1423-925f-40b5-8429-943386b99ef2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1491</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>R5909439</td>\n",
       "      <td>ROSMAP</td>\n",
       "      <td>Human</td>\n",
       "      <td>female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>10407afc-e3e9-4fea-8b37-044eab939981</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1492</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>R8594936</td>\n",
       "      <td>ROSMAP</td>\n",
       "      <td>Human</td>\n",
       "      <td>female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>a8ddcdd4-4be0-4200-bd3c-9a84228bfb6f</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1493</th>\n",
       "      <td>IndividualHumanMetadataTemplate</td>\n",
       "      <td>R2742141</td>\n",
       "      <td>ROSMAP</td>\n",
       "      <td>Human</td>\n",
       "      <td>female</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>single nucleus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>frozen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snrnaSeq</td>\n",
       "      <td>5cb3c8f2-2344-4a5e-82ed-bd4eb9597410</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1194 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Component_x           individualID  cohort species  \\\n",
       "0     IndividualHumanMetadataTemplate  AMPAD_MSSM_0000029668    MSBB   Human   \n",
       "1     IndividualHumanMetadataTemplate  AMPAD_MSSM_00000M1061    MSBB   Human   \n",
       "2     IndividualHumanMetadataTemplate  AMPAD_MSSM_00000M1062    MSBB   Human   \n",
       "3     IndividualHumanMetadataTemplate  AMPAD_MSSM_0000010001    MSBB   Human   \n",
       "4     IndividualHumanMetadataTemplate  AMPAD_MSSM_0000010031    MSBB   Human   \n",
       "...                               ...                    ...     ...     ...   \n",
       "1489  IndividualHumanMetadataTemplate               R5508487  ROSMAP   Human   \n",
       "1490  IndividualHumanMetadataTemplate               R9380629  ROSMAP   Human   \n",
       "1491  IndividualHumanMetadataTemplate               R5909439  ROSMAP   Human   \n",
       "1492  IndividualHumanMetadataTemplate               R8594936  ROSMAP   Human   \n",
       "1493  IndividualHumanMetadataTemplate               R2742141  ROSMAP   Human   \n",
       "\n",
       "         sex                       race               ethnicity  \\\n",
       "0       male  Black or African American  Not Hispanic or Latino   \n",
       "1       male  Black or African American  Not Hispanic or Latino   \n",
       "2     female                        NaN      Hispanic or Latino   \n",
       "3     female                      White  Not Hispanic or Latino   \n",
       "4       male                        NaN      Hispanic or Latino   \n",
       "...      ...                        ...                     ...   \n",
       "1489    male  Black or African American  Not Hispanic or Latino   \n",
       "1490  female  Black or African American  Not Hispanic or Latino   \n",
       "1491  female  Black or African American  Not Hispanic or Latino   \n",
       "1492  female  Black or African American  Not Hispanic or Latino   \n",
       "1493  female  Black or African American  Not Hispanic or Latino   \n",
       "\n",
       "      yearsEducation  apoeGenotype  diagnosis  ...  nucleicAcidSource  \\\n",
       "0                NaN          33.0        NaN  ...     single nucleus   \n",
       "1                NaN          33.0        NaN  ...     single nucleus   \n",
       "2                NaN          33.0        NaN  ...     single nucleus   \n",
       "3                NaN          33.0        NaN  ...     single nucleus   \n",
       "4                NaN          33.0        NaN  ...     single nucleus   \n",
       "...              ...           ...        ...  ...                ...   \n",
       "1489             NaN          33.0        NaN  ...     single nucleus   \n",
       "1490             NaN          23.0        NaN  ...     single nucleus   \n",
       "1491             NaN          44.0        NaN  ...     single nucleus   \n",
       "1492             NaN          33.0        NaN  ...     single nucleus   \n",
       "1493             NaN          33.0        NaN  ...     single nucleus   \n",
       "\n",
       "     cellType samplingAge  samplingAgeUnits  sampleStatus  fastingState  \\\n",
       "0         NaN         NaN               NaN        frozen           NaN   \n",
       "1         NaN         NaN               NaN        frozen           NaN   \n",
       "2         NaN         NaN               NaN        frozen           NaN   \n",
       "3         NaN         NaN               NaN        frozen           NaN   \n",
       "4         NaN         NaN               NaN        frozen           NaN   \n",
       "...       ...         ...               ...           ...           ...   \n",
       "1489      NaN         NaN               NaN        frozen           NaN   \n",
       "1490      NaN         NaN               NaN        frozen           NaN   \n",
       "1491      NaN         NaN               NaN        frozen           NaN   \n",
       "1492      NaN         NaN               NaN        frozen           NaN   \n",
       "1493      NaN         NaN               NaN        frozen           NaN   \n",
       "\n",
       "     visitNumber     assay                                    Id  entityId  \n",
       "0            NaN  snrnaSeq  337399ff-c8cd-4ee2-a87c-11488db02558       NaN  \n",
       "1            NaN  snrnaSeq  bf1ad418-d4bc-492a-8426-b66c8fe7e9a2       NaN  \n",
       "2            NaN  snrnaSeq  50757002-0330-441c-9e61-2b481494bb09       NaN  \n",
       "3            NaN  snrnaSeq  1593a784-36bf-449d-a476-2c6d29f11858       NaN  \n",
       "4            NaN  snrnaSeq  c2514bf6-2229-4a55-a3fe-65f7a673ddd1       NaN  \n",
       "...          ...       ...                                   ...       ...  \n",
       "1489         NaN  snrnaSeq  2b02ebb9-a0e9-4883-9e2e-7ba7350499f7       NaN  \n",
       "1490         NaN  snrnaSeq  b9ad1423-925f-40b5-8429-943386b99ef2       NaN  \n",
       "1491         NaN  snrnaSeq  10407afc-e3e9-4fea-8b37-044eab939981       NaN  \n",
       "1492         NaN  snrnaSeq  a8ddcdd4-4be0-4200-bd3c-9a84228bfb6f       NaN  \n",
       "1493         NaN  snrnaSeq  5cb3c8f2-2344-4a5e-82ed-bd4eb9597410       NaN  \n",
       "\n",
       "[1194 rows x 40 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nps_ad_indv = pd.read_csv(syn.get('syn55251012').path)\n",
    "nps_ad_bios = pd.read_csv(syn.get('syn55251032').path)\n",
    "\n",
    "nps_ad_gold = pd.merge(nps_ad_indv, nps_ad_bios, on=\"individualID\")\n",
    "\n",
    "ros_and_msbb_in_nps_indv = nps_ad_gold[(nps_ad_gold['cohort'] == 'MSBB') | (nps_ad_gold['cohort'] == 'ROSMAP')]\n",
    "ros_and_msbb_in_nps_indv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bdc79417-c4aa-45dd-828b-cabadb98fe14",
   "metadata": {},
   "source": [
    "#### Get WGS assay metadata in MSBB and ROSMAP studies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "04582b2b-4449-433a-9101-50f522150e06",
   "metadata": {},
   "outputs": [],
   "source": [
    "msbb_wgs = pd.read_csv(syn.get('syn22360825').path)\n",
    "rosmap_wgs = pd.read_csv(syn.get('syn21314542').path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a071240-9099-4e10-b196-1ed68ea1f4f8",
   "metadata": {},
   "source": [
    "#### Get snpArray assay metadata in ROSMAP study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "24c2ccce-fa5f-4f54-badb-962ecf80fee2",
   "metadata": {},
   "outputs": [],
   "source": [
    "rosmap_snp = pd.read_csv(syn.get('syn21314550').path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b72fd73-89b7-47cc-a57c-90a9c278e52a",
   "metadata": {},
   "source": [
    "#### For MSBB samples, filter specimenID to last 5 characters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f0c3dd21-00f8-45f5-bd81-612c22ff9208",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WARNING] /tmp/ipykernel_206797/378524507.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ros_and_msbb_in_nps_indv['specimenID'] = np.where(ros_and_msbb_in_nps_indv['cohort'] == 'MSBB',\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ros_and_msbb_in_nps_indv['specimenID'] = np.where(ros_and_msbb_in_nps_indv['cohort'] == 'MSBB', \n",
    "                                                  ros_and_msbb_in_nps_indv['specimenID'].str[-5:], ros_and_msbb_in_nps_indv['cohort'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "48391649-3ab1-4eb1-bd50-636503dfa800",
   "metadata": {},
   "outputs": [],
   "source": [
    "msbb_hits = [str(x) for x in msbb_wgs['specimenID'].astype(str).tolist() if str(x) in ros_and_msbb_in_nps_indv['specimenID'].astype(str).tolist()]\n",
    "rosmap_hits = [str(x) for x in rosmap_wgs['specimenID'].astype(str).tolist() if str(x) in ros_and_msbb_in_nps_indv['specimenID'].astype(str).tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "db88fe3b-7d50-41c4-a871-042e8919668d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rosmap_hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7dda2bd5-bdf3-43e9-ae30-2163afab677c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['71719', '71734', '71789']"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "msbb_hits"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b161fa98-9e45-4134-9913-02604a4677e9",
   "metadata": {},
   "source": [
    "#### 1/17: Find WGS and SNParray data for NPS-AD individuals in MSBB and ROSMAP cohorts by looking in Diverse Cohorts, MSBB, and ROSMAP\n",
    "\n",
    "Have:\n",
    "* NPS-AD individuals in MSBB and ROSMAP cohorts: `ros_and_msbb_in_nps_indv`\n",
    "\n",
    "Need:\n",
    "* MSBB biospecimen metadata:\n",
    "* MSBB WGS assay metadata: `msbb_wgs`\n",
    "* ROSMAP biospecimen metadata:\n",
    "* ROSMAP WGS assay metadata: `rosmap_wgs`\n",
    "* ROSMAP snpArray assay metadata: `rosmap_snp`\n",
    "* AMP-AD_DiverseCohorts biospecimen metadata:\n",
    "* AMP-AD_DiverseCohorts WGS metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "80ecfc15-8a27-4a1c-94c3-8595f443a454",
   "metadata": {},
   "outputs": [],
   "source": [
    "msbb_bio = pd.read_csv(syn.get('syn21893059').path)\n",
    "msbb_wgs = pd.read_csv(syn.get('syn22360825').path)\n",
    "rosmap_bio = pd.read_csv(syn.get('syn21323366').path)\n",
    "ampad_bio = pd.read_csv(syn.get('syn51757645').path)\n",
    "ampad_wgs = pd.read_csv(syn.get('syn51757644').path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10f55660-77ad-4380-9534-6b6b853757b4",
   "metadata": {},
   "source": [
    "#### MSBB biospecimen metadata joined with WGS assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0acc15c4-dfd6-421f-9a38-c4c23911ebaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "msbb_wgs['specimenID'] = msbb_wgs['specimenID'].astype(str)\n",
    "msbb_bio_on_wgs = pd.merge(msbb_bio, msbb_wgs, on=\"specimenID\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "091ff203-f00f-4d87-984b-c26c24083940",
   "metadata": {},
   "source": [
    "#### AMP-AD_DiverseCohorts biospecimen metadata joined with WGS assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1c0f4d34-eb15-43c3-bf21-99263848c932",
   "metadata": {},
   "outputs": [],
   "source": [
    "ampad_bio_on_wgs = pd.merge(ampad_bio, ampad_wgs, on=\"specimenID\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42c82376-b0ae-4572-9245-990e43a8859c",
   "metadata": {},
   "source": [
    "#### ROSMAP biospecimen metadata joined with WGS assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9ce691e1-466f-47b1-825f-c0bbd68f238f",
   "metadata": {},
   "outputs": [],
   "source": [
    "rosmap_bio_on_wgs = pd.merge(rosmap_bio, rosmap_wgs, on=\"specimenID\")\n",
    "rosmap_bio_on_wgs = rosmap_bio_on_wgs[rosmap_bio_on_wgs['assay_x'] == 'wholeGenomeSeq']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b3e5f22-1540-4131-ad22-b2928e87881a",
   "metadata": {},
   "source": [
    "#### ROSMAP biospecimen metadata joined with snpArray assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9da0673d-78b1-48d8-bf90-4a4c8a9d2cd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "rosmap_bio_on_snp = pd.merge(rosmap_bio, rosmap_snp, on=\"specimenID\")\n",
    "rosmap_bio_on_snp = rosmap_bio_on_snp[rosmap_bio_on_snp['assay_x'] == 'snpArray']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dcdf6eb5-6f0d-4759-b06b-be8f3df29b5b",
   "metadata": {},
   "source": [
    "#### NPS-AD individuals in MSBB and ROSMAP cohorts appearing in MSBB WGS assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c1eb858f-004f-41ad-b59c-5968ddfa54a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_msbb_wgs = [x for x in msbb_bio_on_wgs['individualID'].astype(str).tolist() if x in \n",
    "                   ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist()]\n",
    "nps_in_msbb_wgs = list(set(nps_in_msbb_wgs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "2707cb32-d3a6-45f0-9371-09f37dbce839",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('NPS-AD_individuals_in_MSBB_WGS_assay_metadata.txt', 'w+') as f:\n",
    "    for item in nps_in_msbb_wgs:\n",
    "        f.write('%s\\n' %item)\n",
    "\n",
    "f.close()    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e614f551-65ef-4d66-8d16-35c79a22503b",
   "metadata": {},
   "source": [
    "#### NPS-AD individuals in MSBB and ROSMAP cohorts appearing in AMP-AD_DiverseCohorts WGS assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "35f0d31b-8aca-4a53-94cf-e3d8fed0dd65",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_ampad_wgs = [x for x in ampad_bio_on_wgs['individualID'].astype(str).tolist() if x in \n",
    "                   ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist()]\n",
    "nps_in_ampad_wgs = list(set(nps_in_ampad_wgs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "bd10297d-e59f-4b09-a2e4-1769a743e1e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('NPS-AD_individuals_in_AMP-AD_DiverseCohorts_WGS_assay_metadata.txt', 'w+') as f:\n",
    "    for item in nps_in_ampad_wgs:\n",
    "        f.write('%s\\n' %item)\n",
    "\n",
    "f.close()    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f29be72-cf93-43f8-af47-ee19f8cd6d3a",
   "metadata": {},
   "source": [
    "#### NPS-AD individuals in MSBB and ROSMAP cohorts appearing in ROSMAP WGS assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f2e4e52a-10d8-46c1-b015-f2f1922e33ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_rosmap_wgs = [x for x in rosmap_bio_on_wgs['individualID'].astype(str).tolist() if x in \n",
    "                   ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist()]\n",
    "nps_in_rosmap_wgs = list(set(nps_in_rosmap_wgs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "8762aa15-3761-40e4-8657-806c12e6e53b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('NPS-AD_individuals_in_ROSMAP_WGS_assay_metadata.txt', 'w+') as f:\n",
    "    for item in nps_in_rosmap_wgs:\n",
    "        f.write('%s\\n' %item)\n",
    "\n",
    "f.close() "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09dc8716-8e3a-496d-98ec-f97104150100",
   "metadata": {},
   "source": [
    "#### NPS-AD individuals in MSBB and ROSMAP cohorts appearing in ROSMAP snpArray assay metadata:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "06758557-6ba8-4f1d-9fe1-0be278e7df1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_rosmap_snp = [x for x in rosmap_bio_on_snp['individualID'].astype(str).tolist() if x in \n",
    "                   ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist()]\n",
    "nps_in_rosmap_snp = list(set(nps_in_rosmap_snp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "0366d25b-1160-4a63-bb68-4ebdec4b2cd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('NPS-AD_individuals_in_ROSMAP_snpArray_assay_metadata.txt', 'w+') as f:\n",
    "    for item in nps_in_rosmap_snp:\n",
    "        f.write('%s\\n' %item)\n",
    "\n",
    "f.close() "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0cd22e9-5a0c-4d67-94ac-a8901c1a58aa",
   "metadata": {},
   "source": [
    "#### 1/24: Create new NPS-AD metadata file to summarize WGS and snpArray data existing for NPS-AD individuals that appear in other studies\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7a34987-8bee-4abb-b1ae-81b01d9ea0d6",
   "metadata": {},
   "source": [
    "#### Joined biospecimen and WGS assay metadata for NPS-AD individuals in the MSBB study:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "00f1f1ed-d863-4b4f-a8ed-e2744f056f87",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WARNING] /tmp/ipykernel_206797/125951702.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  nps_in_msbb_wgs_full['study'] = 'MSBB'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "nps_in_msbb_wgs_full = msbb_bio_on_wgs[msbb_bio_on_wgs['individualID'].isin(ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist())]\n",
    "nps_in_msbb_wgs_full['study'] = 'MSBB'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "08919d1d-558e-4915-ac09-ec8ec4232dca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['individualID', 'specimenID', 'specimenIdSource', 'organ', 'tissue',\n",
       "       'BrodmannArea', 'sampleStatus', 'tissueWeight', 'tissueVolume',\n",
       "       'nucleicAcidSource', 'cellType', 'fastingState', 'assay_x',\n",
       "       'isPostMortem', 'samplingAge', 'visitNumber', 'exclude',\n",
       "       'excludeReason', 'samplingAgeUnits', 'assay_y', 'platform', 'dnaBatch',\n",
       "       'sequencingBatch', 'libraryPrep', 'libraryPreparationMethod',\n",
       "       'readLength', 'runType', 'totalReads', 'numberCells', '260/280',\n",
       "       '260/230', 'GQN', 'study'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nps_in_msbb_wgs_full.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "64129dc9-d4fa-4b91-82ba-caf0c9b07d24",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_msbb_wgs_full = nps_in_msbb_wgs_full[['individualID', 'specimenID', \n",
    "                                             'organ', 'tissue', 'BrodmannArea', 'sampleStatus',\n",
    "                                             'nucleicAcidSource', 'assay',\n",
    "                                             'isPostMortem', 'dnaBatch', 'study']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3cb79a97-f498-4512-84e6-a181e615d040",
   "metadata": {},
   "source": [
    "#### Joined biospecimen and WGS assay metadata for NPS-AD individuals in the AMP-AD_Diverse_Cohorts study:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1bd15d3e-28b9-4a59-b2d6-238d76885197",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WARNING] /tmp/ipykernel_206797/1738449393.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  nps_in_ampad_wgs_full['study'] = 'AMP-AD_DiverseCohorts'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "nps_in_ampad_wgs_full = ampad_bio_on_wgs[ampad_bio_on_wgs['individualID'].isin(ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist())]\n",
    "nps_in_ampad_wgs_full['study'] = 'AMP-AD_DiverseCohorts'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "1fdb70a7-8242-4952-879b-7a6108743d80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['individualID', 'specimenID', 'dataGenerationSite', 'organ', 'tissue',\n",
       "       'BrodmannArea', 'isPostMortem', 'sampleStatus', 'nucleicAcidSource',\n",
       "       'assay', 'specimenMetadataSource', 'NYGC Sample Name', 'Batch',\n",
       "       'Concentration (ng/uL)', 'Mass (ng)', 'GQN', 'Container Name', 'Well',\n",
       "       'Site', 'Library Prep', 'Library Prep Plate', 'Comments', 'Project',\n",
       "       'Total Reads', 'PF Reads', 'Percent PF Reads', 'PF Aligned Reads',\n",
       "       'Percent PF Aligned', 'PF Aligned Pairs', 'Percent PF Aligned Pairs',\n",
       "       'Mean Read Length', 'Strand Balance', 'Estimated Library Size',\n",
       "       'Mean Coverage', 'Percent Sequence Contamination', 'Median Insert Size',\n",
       "       'Mean Insert Size', 'AT Dropout', 'GC Dropout', 'Percent Total Dup.',\n",
       "       'study'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nps_in_ampad_wgs_full.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "14b341b6-1d8a-439a-b5af-84241d7d95ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_ampad_wgs_full = nps_in_ampad_wgs_full[['individualID', 'specimenID', \n",
    "                                             'organ', 'tissue', 'BrodmannArea', 'sampleStatus',\n",
    "                                             'nucleicAcidSource', 'assay',\n",
    "                                             'isPostMortem', 'dnaBatch', 'study']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4a9cd16-72d9-4f14-839d-6126e5738422",
   "metadata": {},
   "source": [
    "#### Joined biospecimen and WGS assay metadata for NPS-AD individuals in the ROSMAP study:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "2ee19fae-8b60-493d-b9d9-c1b0f82c739b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WARNING] /tmp/ipykernel_206797/1204183405.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  nps_in_rosmap_wgs_full['study'] = 'ROSMAP'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "nps_in_rosmap_wgs_full = rosmap_bio_on_wgs[rosmap_bio_on_wgs['individualID'].isin(ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist())]\n",
    "nps_in_rosmap_wgs_full['study'] = 'ROSMAP'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "1efaa632-b567-4458-97c3-5c36f9b71a88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['individualID', 'specimenID', 'specimenIdSource', 'organ', 'tissue',\n",
       "       'BrodmannArea', 'sampleStatus', 'tissueWeight', 'tissueVolume',\n",
       "       'nucleicAcidSource', 'cellType', 'fastingState', 'isPostMortem',\n",
       "       'samplingAge', 'samplingAgeUnits', 'visitNumber', 'assay_x', 'exclude',\n",
       "       'excludeReason', 'samplingDate', 'assay_y', 'platform', 'dnaBatch',\n",
       "       'sequencingBatch', 'libraryID', 'libraryPrep',\n",
       "       'libraryPreparationMethod', 'readLength', 'runType', 'totalReads',\n",
       "       'numberCells', 'X260.280', 'X260.230', 'GQN', 'referenceSet', 'study'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nps_in_rosmap_wgs_full.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "cd1e73ae-9ec1-4899-84b6-e078af6c6de0",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_rosmap_wgs_full = nps_in_rosmap_wgs_full[['individualID', 'specimenID', \n",
    "                                             'organ', 'tissue', 'BrodmannArea', 'sampleStatus',\n",
    "                                             'nucleicAcidSource', 'assay',\n",
    "                                             'isPostMortem', 'dnaBatch', 'study']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57636eaf-dac5-4d2c-87e3-c017b6459143",
   "metadata": {},
   "source": [
    "#### Joined biospecimen and snpArray assay metadata for NPS-AD individuals in the ROSMAP study:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "5d38e10f-9619-44fc-8247-5c285577ea6e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WARNING] /tmp/ipykernel_206797/2114837877.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  nps_in_rosmap_snp_full['study'] = 'ROSMAP'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "nps_in_rosmap_snp_full = rosmap_bio_on_snp[rosmap_bio_on_snp['individualID'].isin(ros_and_msbb_in_nps_indv['individualID'].astype(str).tolist())]\n",
    "nps_in_rosmap_snp_full['study'] = 'ROSMAP'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "8899a7c5-748d-4fc4-aec5-56f5cb8fa0ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['individualID', 'specimenID', 'specimenIdSource', 'organ', 'tissue',\n",
       "       'BrodmannArea', 'sampleStatus', 'tissueWeight', 'tissueVolume',\n",
       "       'nucleicAcidSource', 'cellType', 'fastingState', 'isPostMortem',\n",
       "       'samplingAge', 'samplingAgeUnits', 'visitNumber', 'assay_x', 'exclude',\n",
       "       'excludeReason', 'samplingDate', 'assay_y', 'platform', 'dnaBatch',\n",
       "       'arrayBatch', '260/280', '260/230', 'GQN', 'study'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nps_in_rosmap_snp_full.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "8b5c4688-347c-48e2-a188-23a6a659cc32",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_in_rosmap_snp_full = nps_in_rosmap_snp_full[['individualID', 'specimenID', \n",
    "                                             'organ', 'tissue', 'BrodmannArea', 'sampleStatus',\n",
    "                                             'nucleicAcidSource', 'assay',\n",
    "                                             'isPostMortem', 'dnaBatch', 'study']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce263b13-a8e4-490e-b078-3c97ebaa0282",
   "metadata": {},
   "source": [
    "#### Concatenate all 4 files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "dc7fd9f2-ee28-4bad-8875-3548f5293c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_ad_wgs_snp = pd.concat([nps_in_msbb_wgs_full, nps_in_ampad_wgs_full, nps_in_rosmap_wgs_full, nps_in_rosmap_snp_full], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "184dba08-dde4-4cee-9540-a20e061a6b4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>specimenID</th>\n",
       "      <th>organ</th>\n",
       "      <th>tissue</th>\n",
       "      <th>BrodmannArea</th>\n",
       "      <th>sampleStatus</th>\n",
       "      <th>nucleicAcidSource</th>\n",
       "      <th>assay</th>\n",
       "      <th>isPostMortem</th>\n",
       "      <th>dnaBatch</th>\n",
       "      <th>study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AMPAD_MSSM_0000026992</td>\n",
       "      <td>68411</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wholeGenomeSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSBB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AMPAD_MSSM_0000050580</td>\n",
       "      <td>68416</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wholeGenomeSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSBB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AMPAD_MSSM_0000084538</td>\n",
       "      <td>68419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wholeGenomeSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSBB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMPAD_MSSM_0000077591</td>\n",
       "      <td>68420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wholeGenomeSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSBB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AMPAD_MSSM_0000029716</td>\n",
       "      <td>68422</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wholeGenomeSeq</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSBB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>R5009796</td>\n",
       "      <td>ROS21246218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snpArray</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ROSMAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>R4689636</td>\n",
       "      <td>ROS10288185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snpArray</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ROSMAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>R4184741</td>\n",
       "      <td>ROS11624423</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snpArray</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ROSMAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1008</th>\n",
       "      <td>R2845830</td>\n",
       "      <td>ROS10424475</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snpArray</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ROSMAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>R2323911</td>\n",
       "      <td>MAP50101497</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snpArray</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ROSMAP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>454 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               individualID   specimenID organ tissue BrodmannArea  \\\n",
       "0     AMPAD_MSSM_0000026992        68411   NaN    NaN          NaN   \n",
       "1     AMPAD_MSSM_0000050580        68416   NaN    NaN          NaN   \n",
       "2     AMPAD_MSSM_0000084538        68419   NaN    NaN          NaN   \n",
       "3     AMPAD_MSSM_0000077591        68420   NaN    NaN          NaN   \n",
       "4     AMPAD_MSSM_0000029716        68422   NaN    NaN          NaN   \n",
       "...                     ...          ...   ...    ...          ...   \n",
       "994                R5009796  ROS21246218   NaN    NaN          NaN   \n",
       "997                R4689636  ROS10288185   NaN    NaN          NaN   \n",
       "998                R4184741  ROS11624423   NaN    NaN          NaN   \n",
       "1008               R2845830  ROS10424475   NaN    NaN          NaN   \n",
       "1010               R2323911  MAP50101497   NaN    NaN          NaN   \n",
       "\n",
       "     sampleStatus nucleicAcidSource           assay isPostMortem dnaBatch  \\\n",
       "0             NaN               NaN  wholeGenomeSeq         True      NaN   \n",
       "1             NaN               NaN  wholeGenomeSeq         True      NaN   \n",
       "2             NaN               NaN  wholeGenomeSeq         True      NaN   \n",
       "3             NaN               NaN  wholeGenomeSeq         True      NaN   \n",
       "4             NaN               NaN  wholeGenomeSeq         True      NaN   \n",
       "...           ...               ...             ...          ...      ...   \n",
       "994           NaN               NaN        snpArray          NaN      NaN   \n",
       "997           NaN               NaN        snpArray          NaN      NaN   \n",
       "998           NaN               NaN        snpArray          NaN      NaN   \n",
       "1008          NaN               NaN        snpArray          NaN      NaN   \n",
       "1010          NaN               NaN        snpArray          NaN      NaN   \n",
       "\n",
       "       study  \n",
       "0       MSBB  \n",
       "1       MSBB  \n",
       "2       MSBB  \n",
       "3       MSBB  \n",
       "4       MSBB  \n",
       "...      ...  \n",
       "994   ROSMAP  \n",
       "997   ROSMAP  \n",
       "998   ROSMAP  \n",
       "1008  ROSMAP  \n",
       "1010  ROSMAP  \n",
       "\n",
       "[454 rows x 11 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nps_ad_wgs_snp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "cd54076f-a0c5-4fcd-b5aa-b57db3fe3548",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "293"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nps_in_msbb_wgs_full)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "a12b18fe-f996-4d6e-be50-d0bfee2d7970",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "151"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nps_in_ampad_wgs_full)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "083c30bc-425e-4e79-acda-6be3b4f51ee0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nps_in_rosmap_wgs_full)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "a552a8ee-5a5c-4510-8ef6-9b5cc1339ccc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nps_in_rosmap_snp_full)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "cabfad53-3866-4a46-8f2b-080649605e95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "454"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "293+151+1+9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "fe10e12d-dda4-4bfc-8457-fa97ea38d42a",
   "metadata": {},
   "outputs": [],
   "source": [
    "nps_ad_wgs_snp.to_csv('NPS-AD_individuals_with_WGS_and_snpArray_metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "753ec851-3e65-42e8-b603-e3ea05880400",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
